Clairified collections - closes #612 (#644)
ietf-tapswg/api-drafts: draft-ietf-taps-interface.md

  selection; see {{selection-props}}. When querying, a Preference is of
  type Boolean, with `true` indicating that the Selection Property
  has been applied.

> - Tuple: A specific grouping of multiple value types.

Reviewer: ```suggestion
- Tuple: An ordered grouping of multiple value types.
```
Unless "specific" really means something here? But I wouldn't know what it means.

> - Tuple: takes values composed from types defined above.

Reviewer 2: A specific grouping of multiple value types.

  In this this document, it is written as a list in brackets, e.g., ```(Enumeration, Preference)``` 

>   The compsition of types and the order depends on the property and is fix for the property.

Reviewer 2: ```suggestion
  The composition of types and the order depends on the property and is fix for the property.
```

>   The composition of types and the order depends on the property and is fixed for the property.

Reviewer: ```suggestion
  The composition of types and their order depends on the property and is fixed for the property.
```

>   The actual representation is implementation-dependen.

Reviewer: ```suggestion
  The actual representation is implementation-dependent.
```

> - Collection: A group of one or more value of the same type.

Reviewer: ```suggestion
- Collection: An unordered grouping of one or more values of the same type.
```
This way, it's consistent with Tuple being a "grouping" as well.

> - Collection: takes a valiable number of a single of the above types.

Reviewer 2: ```suggestion
- Collection: A group of one or more value types.
```

Author: I like your wording but would prefer to stress that these have to be of the same type.

>   The actual representation, e.g. as a set or an array, is implementation-dependent.

Reviewer: Not sure we need this sentence. It basically says "The implementation is implementation-dependent", doesn't it? 
Also, a set is not necessarily ordered, while an array is usually ordered, I would say. So I think removing this sentence would make the entire thing less confusing.

Author: I still see value in the sentence – especially stressing that the grouping is by design unordered, but might be implemented with some ordered data structure.

Reviewer: I'm not sure why it'd be necessary to stress how it could be implemented in the interface draft.
But I'm fine with merging this PR as it is.

>   The actual representation, e.g. as set or array, is implementation-dependen.

Reviewer 2: ```suggestion
  The actual representation, e.g. as a set or an array, is implementation-dependent.
```
